pkg/volume/azure_dd/azure_provision.go

```diff
@@ -137,6 +169,19 @@ func (p *azureDiskProvisioner) Provision(selectedNode *v1.Node, allowedTopologie
 		return nil, err
 	}
 
+	zoned, err = parseZoned(strZoned, kind)
+	if err != nil {
+		return nil, err
+	}
+
+	if !zoned && (zonePresent || zonesPresent) {
+		return nil, fmt.Errorf("zone or zones StorageClass parameters must be used together with zoned parameter")
+	}
+
+	if zonePresent && zonesPresent {
+		return nil, fmt.Errorf("both zone and zones StorageClass parameters must not be used at the same time")
+	}
+
```

Comment:
The zone parameters can also be specified through `allowedTopologies`. I am adding a volumeutils.PopulateZoneConfig() that will look at all the parameters: zone/zones/allowedTopologies and return a list of zones as well as do all the validation necessary. Once that is ready, we can switch to use that here and remove the validations above.

Reply:
@ddebroy Thanks. allowedTopologies will be added in the next PR.

Reply:
I would suggest only supporting a "zones" argument.  The reason aws and gce have both parameters was because zones was added later and we couldn't break backwards compatibility with zone.  But since this is new for azure, I think the code can be simplified if you only support zones.